src/components/ReactComponent.jsx

```diff
@@ -114,8 +115,9 @@ export default class ReactComponent extends Field {
    * The user has changed the value in the component and the value needs to be updated on the main submission object and other components notified of a change event.
    *
    * @param value
+   * @param flags
    */
-  updateValue = (value, flags) => {
+  updateValue = (value, flags = []) => {
```

Comment:
Flags must be an object, not an array. Line 121 defines flags` default value, why do we need to have it here as well?